tests/cppunit/string_util_test.cc

```diff
@@ -84,6 +84,119 @@ TEST(StringUtil, HasPrefix) {
   ASSERT_FALSE(util::HasPrefix("has", "has_prefix"));
 }
 
+TEST(StringUtil, StringMatch) {
+  /* Some basic tests */
+  EXPECT_TRUE(util::StringMatch("a", "a"));
+  EXPECT_FALSE(util::StringMatch("a", "b"));
+  EXPECT_FALSE(util::StringMatch("a", "aa"));
+  EXPECT_FALSE(util::StringMatch("a", ""));
+  EXPECT_TRUE(util::StringMatch("", ""));
+  EXPECT_FALSE(util::StringMatch("", "a"));
+  EXPECT_TRUE(util::StringMatch("*", ""));
+  EXPECT_TRUE(util::StringMatch("*", "a"));
+
+  /* Simple character class tests */
+  EXPECT_TRUE(util::StringMatch("[a]", "a"));
+  EXPECT_FALSE(util::StringMatch("[a]", "b"));
+  EXPECT_FALSE(util::StringMatch("[^a]", "a"));
+  EXPECT_TRUE(util::StringMatch("[^a]", "b"));
+  EXPECT_TRUE(util::StringMatch("[ab]", "a"));
+  EXPECT_TRUE(util::StringMatch("[ab]", "b"));
+  EXPECT_FALSE(util::StringMatch("[ab]", "c"));
+  EXPECT_TRUE(util::StringMatch("[^ab]", "c"));
+  EXPECT_TRUE(util::StringMatch("[a-c]", "b"));
+  EXPECT_FALSE(util::StringMatch("[a-c]", "d"));
```

Comment:
I'm wondering what will happen if user inputs something like `[a`, or `[a-]` i.e. wrong glob patterns.

Maybe we can do some syntax check for glob patterns? although it isn't required to be added in this patch.

Reply:
Added some code to validate globs and return error messages on invalid globs